src/project-scanner.ts

```diff
@@ -73,21 +76,59 @@ export class ProjectScanner {
     return packageScanner.getConversionManifest();
   }
 
+  /**
+   * Reads the entrypoints for a package from the `main` field in the package's
+   * 'bower.json' file.
+   */
+  private async readBowerJsonEntrypoints(packageName: string) {
+    const bowerJsonPath =
+        path.join(this.urlHandler.getPackageDir(packageName), 'bower.json');
+
+    let bowerJson: Partial<BowerConfig>|undefined = undefined;
+    try {
+      bowerJson = await fse.readJSON(bowerJsonPath) as Partial<BowerConfig>;
+    } catch {
+      console.warn(
+          `Failed to read entrypoints of package '${packageName}' because ` +
+          `the package does not have a 'bower.json' file.`);
+    }
+
+    let bowerMainFiles = bowerJson ? bowerJson.main : [];
+    if (typeof bowerMainFiles === 'string') {
+      bowerMainFiles = [bowerMainFiles];
+    } else if (bowerMainFiles === undefined) {
+      bowerMainFiles = [];
+    }
+
+    return bowerMainFiles.map((relativeOriginalUrl) => {
+      return this.urlHandler.packageRelativeToOriginalUrl(
```

Comment:
we'll usually remove the `return` statement when we use a single-line callback.

```
return bowerMainFiles.map((relativeOriginalUrl) => 
  this.urlHandler.packageRelativeToOriginalUrl(packageName, relativeOriginalUrl));
```

Reply:
fixed